### PR TITLE
Moblity refactor: hands blocked and restrained edition.

### PR DESCRIPTION
**Pull request by Rohesie**
* Splits the `restrained()` proc into component traits: `TRAIT_HANDS_BLOCKED` for the general inability to use hands and `TRAIT_RESTRAINED` for the more specific condition that permits arrests.
* Code moved away from the `update_mobility()` proc so it doesn't have to wait for an update, instead changing based on events. The idea is to eventually kill that proc.
* Wrapper proc added for setting the `handcuffed` value so we can react to the event of it changing.
* Kills the `RestrainedClickOn()` proc. That is now just an `UnarmedAttack()`, in where the ability to use hands can be checked. Monkeys keep their bite attack and humans their self-examine.

### CODE REVIEW

**Comment by bobbahbrown** [comment]
Can you convert the comments that you've added for your defines to dmdoc-compliant comments? (``/// blah \n DEFINE``)

**Comment by Rohesie** [comment]
> 
> 
> Can you convert the comments that you've added for your defines to dmdoc-compliant comments? (`/// blah \n #define`)

Could you suggest changes to show me the right format?

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -79,6 +79,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_KNOCKEDOUT		"knockedout" //Forces the user to stay unconscious.
 #define TRAIT_IMMOBILIZED		"immobilized" //Prevents voluntary movement.
 #define TRAIT_FLOORED			"floored" //Prevents standing or staying up on its own.
+#define TRAIT_HANDS_BLOCKED		"handsblocked" //Prevents usage of manipulation appendages (picking, holding or using items, manipulating storage).
+#define TRAIT_RESTRAINED		"restrained" //Abstract condition that prevents movement if being pulled and might be resisted against. Handcuffs and straight jackets, basically.
```
Comment (bobbahbrown) [inline code/__DEFINES/traits.dm:83]:
```suggestion
/// Prevents usage of manipulation appendages (picking, holding or using items, manipulating storage).
#define TRAIT_HANDS_BLOCKED		"handsblocked"
/// Abstract condition that prevents movement if being pulled and might be resisted against. Handcuffs and straight jackets, basically.
#define TRAIT_RESTRAINED		"restrained"
```

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -286,6 +288,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define RESTING_TRAIT "resting" //trait associated to resting
 #define STAT_TRAIT "stat" //trait associated to a stat value or range of
 #define MAPPING_HELPER_TRAIT "mapping-helper" //obtained from mapping helper
+#define SUIT_TRAIT "suit" //associated to wearing a suit
```
Comment (bobbahbrown) [inline code/__DEFINES/traits.dm:291]:
```suggestion
/// Trait associated to wearing a suit
#define SUIT_TRAIT "suit"
```

**Review by bobbah — COMMENTED** [review]
as requested, you can see more information in the [dmdoc requirements](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-variabledefine) -- I'll be honest here it doesn't look like the language server currently consumes the comment but it's something that is definitely planned, so having the comments in the format which it consumes will be super helpful for tooltips and other visuals

File: code/_onclick/other_mobs.dm [hunk 3]
```diff
@@ -103,74 +100,65 @@
...
+		if(a_intent != INTENT_HARM || is_muzzled())
+			return
+		if(!iscarbon(A))
+			return
+		var/mob/living/carbon/victim = A
+		var/dam_zone = pick(BODY_ZONE_CHEST, BODY_ZONE_PRECISE_L_HAND, BODY_ZONE_PRECISE_R_HAND, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+		var/obj/item/bodypart/affecting = null
+		if(ishuman(victim))
+			var/mob/living/carbon/human/human_victim = victim
+			affecting = human_victim.get_bodypart(ran_zone(dam_zone))
```
[the remark this answers is not in the document]
Comment (Rohesie) [reply code/_onclick/other_mobs.dm:118]: Yea, this was copypasted code I didn't even pause to check it.
Removed the second randomization, it makes no sense.

File: code/datums/action.dm [hunk 4]
```diff
@@ -94,14 +94,12 @@
 /datum/action/proc/IsAvailable()
 	if(!owner)
 		return FALSE
-	if(check_flags & AB_CHECK_RESTRAINED)
-		if(owner.restrained())
+	if(check_flags & AB_CHECK_HANDS_BLOCKED)
+		if(HAS_TRAIT(owner, TRAIT_HANDS_BLOCKED))
+			return FALSE
```
[the remark this answers is not in the document]
Comment (Rohesie) [reply code/datums/action.dm]: Mhm, I was just mass-replacing individual checks without giving each much attention.
Fixed.

**Comment by Rohesie** [comment]
Change requests done.

**Review by Timberpoes — APPROVED** [review]

**Comment by techno0** [comment]
AI needs to spin this is cringe

**Pull request merged**